crates/matrix-sdk-crypto/src/gossiping/machine.rs

```diff
@@ -870,8 +870,11 @@ impl GossipMachine {
         info: &GossipRequest,
         sender_key: Curve25519PublicKey,
         event: &DecryptedForwardedRoomKeyEvent,
+        should_trust_safe_flag: bool,
```

Comment:
I don't think that this is useful, if we trust the claimed sender key then we trust the claimed trust flag as well. If we don't trust either of them, we shouldn't accept the `m.fowrarded_room_key`.

Reply:
I think that with 'share keys on invite' we will receive forwards from another user, and for those we can't trust the claimed_flag

Reply:
Well ok, but there's two problems to this:

1. Please keep it topical, such a change isn't really related to this PR since it doesn't add support for the 'forward keys on invite' feature.
2. I don't think the 'forward keys on invite' feature will use a `m.forwarded_room_key` event and thus the new event type will not have a `trusted` flag at all.